tests/workflows/environment_variable_node_inputs/tests/test_workflow.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+from vellum.workflows.references.environment_variable import EnvironmentVariableReference
+
+from tests.workflows.environment_variable_node_inputs.workflow import (
+    EnvironmentVariableNodeInputsWorkflow,
+    NodeWithEnvironmentVariable,
+)
+
+
+def test_environment_variable_resolves_correctly():
+    """
+    Tests that environment variables resolve to their actual values during node execution.
+    """
+
+    os.environ["TEST_API_KEY"] = "secret-key-12345"
+
+    workflow = EnvironmentVariableNodeInputsWorkflow()
+
+    # WHEN the workflow is run
+    terminal_event = workflow.run()
```

Comment:
DevinAI for this test, can we use the `stream()` method instead, and add assertions for both the node event and the node event with `model_dump()`